### ISSUE TITLE
Saving plan not working

### ISSUE THREAD

**Issue by fosterlynn** [Carbon-Farm-Network/app-carbon-farm-network#94]
![Screenshot from 2024-02-16 14-22-24](https://github.com/Carbon-Farm-Network/app-carbon-farm-network/assets/3776081/829a11ab-41c6-41d4-a1a2-03eba494f7f3)

I did several things, including (from memory) changing quantity on commitment, adding events, moving commitments up and down.  [EDIT: before saving the first time, I added a satisfy request item, which created more commitments, one of which had no offers to start it off.  Just trying to think of all the conditions....]

**Comment by fosterlynn** [comment]
Trying changes one at a time.

Changed one order (top one on the first column).  Probably had cost.
![Screenshot from 2024-02-16 15-28-19](https://github.com/Carbon-Farm-Network/app-carbon-farm-network/assets/3776081/6dfecb68-e1d4-4189-b76b-5b0efd852310)

**Comment by LeosPrograms** [comment]
Those console errors have to do with the drag-to-scroll library, not saving. Did the plan hang on 0 out of 29 indefinitely?

**Comment by fosterlynn** [comment]
>Did the plan hang on 0 out of 29 indefinitely?

Yes.

**Comment by LeosPrograms** [comment]
I suspect this is related to issue #92--if there are no units, resource specifications etc, the plan cannot save. The solution is probably to load those things early and store them, and disallow planning unless they're present.

**Comment by fosterlynn** [comment]
Unofficial version acfn-1.0.4

Still having trouble saving the plan after it was saved once.  It now has everything it needs on the modals, afaict.  There are some messages about classifiedAs; the only thing I can think of that might use classifiedAs is the agent network role, but I'm not sure exactly where that lives.  Same thing on my and Bob's computer.  I have fixed any possible missing data on the cost side, adding offers.  I have tried making changes, and making no changes, just saving.  A few print screens, in reverse chronological order:

made some changes:
![Screenshot from 2024-02-20 16-09-04](https://github.com/Carbon-Farm-Network/app-carbon-farm-network/assets/3776081/72953e46-acf4-4c68-b1b3-32754feec7c8)
made no changes:
![Screenshot from 2024-02-20 15-41-55](https://github.com/Carbon-Farm-Network/app-carbon-farm-network/assets/3776081/3f990dad-9561-4e5c-9d0a-27a9ddb7857a)
